### PR TITLE
Implement each functionality

### PR DESCRIPTION
**Pull request by berry-langerak**
## What

This PR introduces each functionality to Particle\Validator. It will allow users to validate repeating nested arrays.

## How to test

Check the unit tests.

### CODE REVIEW

File: src/Failure.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
+<?php
+namespace Particle\Validator;
+
+class Failure
```
Comment (RickvdStaaij) [inline src/Failure.php]: Wondering why you called this class `Failure`, `Error` feels more appropriate if you ask me.

File: src/Failure.php [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+     */
+    protected $template;
+
+    /**
+     * @var array
+     */
+    protected $parameters = [];
+
+    /**
+     * @param $key
```
Comment (RickvdStaaij) [inline src/Failure.php]: missing type

File: src/Rule/Each.php [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        }
+
+        $result = true;
+        foreach ($value as $index => $innerValue) {
+            $result = $this->validateValue($index, $innerValue) && $result;
+        }
+        return $result;
+    }
+
+    protected function validateValue($index, $value)
```
Comment (RickvdStaaij) [inline src/Rule/Each.php]: Phpdoc missing

File: src/Failure.php (hunk 1 above)
Comment (berry-langerak) [inline src/Failure.php]: `Error` implies that something is *critically* wrong. To me, `Failure` implies that something went wrong but certainly not critical, and certainly expected. After all, we're building a validation library, so having an invalid value is not exceptional: it's what the library is there for ;)

**Comment by RickvdStaaij** [comment]
Can you add a test where you test the successful usage on `each` with multiple values?

**Comment by RickvdStaaij** [comment]
Some comments, but overall looking really good, nice!

**Comment by RickvdStaaij** [comment]
:+1: from me!

**Comment by berry-langerak** [comment]
Merging, then :)

**Pull request merged**